src/jaybenne/sourcing.cpp

```diff
@@ -87,12 +101,44 @@ TaskStatus SourcePhotons(T *md, const Real t_start, const Real dt) {
               [[maybe_unused]] const Real &sbd = sb;
               [[maybe_unused]] const Real &vvd = vv;
               [[maybe_unused]] const Real &dtd = dt;
-              [[maybe_unused]] auto &opac = opacity;
+              [[maybe_unused]] auto opac = opacity;
+              [[maybe_unused]] auto mopac = mopacity;
+              [[maybe_unused]] const auto numind = numin;
+              [[maybe_unused]] const auto numaxd = numax;
+              [[maybe_unused]] const auto n_nubinsd = n_nubins;
               Real erad = 0.0;
               if constexpr (ST == SourceType::thermal) {
                 erad = (4.0 * sbd / vvd) * std::pow(temp, 4.0) * dv;
               } else if constexpr (ST == SourceType::emission) {
-                const Real emis = opac.Emissivity(rho, temp);
+                Real emis = -1.;
+                if constexpr (FT == FrequencyType::gray) {
+                  emis = mopac.Emissivity(rho, temp);
+                } else if constexpr (FT == FrequencyType::multigroup) {
+                  // Construct emission CDF
+                  const Real dlnu = (std::log(numaxd) - std::log(numind)) / n_nubinsd;
+                  Real nu = std::exp(std::log(numind) + 0.5 * dlnu);
+                  Real dnu = nu * dlnu;
```

Comment:
Sorry if missing something here - does `dlnu` need to be exponentiated here to get the frequency group width?, E.g. something like:
```suggestion
                  Real dnu = nu * (std::exp(dlnu) - 1.0);
```

Reply:
I think we're allowed to use `dx = x d log x`

Reply:
Ah - right - good point. However, these formulae only agree at small `dlnu` - for coarse group widths would there be an issue covering the intended range of frequencies?  Then again, the absolute error between these is only O(`dlnu^3`) when the midpoint `nu` is used in your formula, as is true here, with a relative error of O(`dlnu^2`). This implies the relative discrepancy between group widths for these formulae should only be 1% when 10 groups are used, which is maybe not worth worrying about... sorry was just curious - thanks!